### PR TITLE
Add OCP IPMI OEM set/get boot order support

### PR DESCRIPTION
**Pull request by johnnylinwiwynn**
When VPD variable bmc_bootorder_override is set to 1, IPMI OEM
set boot order can override and update VPD boot order, default
is not set so boot order is determined from VPD system booter
configuration alone.
For now IPMI OEM set boot order only supports IPv6 network boot
and local GRUB SSD/HD boot.

Signed-off-by: Johnny Lin <johnny_lin@wiwynn.com>

### CODE REVIEW

**Comment by johnnylinwiwynn** [comment]
@insomniacslk  @jonzhang-fb

File: cmds/boot/systemboot/main.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ var defaultBootsequence = [][]string{
 	{"localboot", "-grub"},
 }
 
+// VPD variable for enabling IPMI BMC overriding boot order, default is not set
+const VpdBmcBootOrderOverride = "bmc_bootorder_override"
+
+var bmcBootOverride bool = false
```
Comment (rminnich) [inline cmds/boot/systemboot/main.go:37]: the zero value of a bool is false, so you might consider dropping the = false part.

File: cmds/boot/systemboot/main.go [hunk 2]
```diff
@@ -97,6 +102,18 @@ func runIPMICommands() {
 	} else {
 		log.Printf("Watchdog is stopped.")
 	}
+	// Try RW_VPD first
+	value, err := systembooter.Get(VpdBmcBootOrderOverride, false)
+	if err != nil {
+		// Try RO_VPD
+		value, err = systembooter.Get(VpdBmcBootOrderOverride, true)
+	}
+	if err == nil && string(value) == "1" {
```
Comment (rminnich) [inline cmds/boot/systemboot/main.go:111]: Could you simply do this: 
if err == nil {
bmcBootOverride = true
}
log.Printf("VPD %s is %v", VpdBmcBootOrderOverride, value)

**Review by rminnich — COMMENTED** [review]
seems fine, I just had two little nits.

File: cmds/boot/systemboot/main.go (hunk 1 above)
Comment (johnnylinwiwynn) [reply cmds/boot/systemboot/main.go]: done.

File: cmds/boot/systemboot/main.go (hunk 2 above)
Comment (johnnylinwiwynn) [reply cmds/boot/systemboot/main.go:111]: done.

**Review by rminnich — APPROVED** [review]

**Review by rminnich — APPROVED** [review]

**Review by rminnich — APPROVED** [review]

**Pull request merged**